kratos/python_scripts/assign_vector_by_direction_process.py

```diff
@@ -69,13 +69,30 @@ def __init__(self, Model, settings ):
 
         self.model_part = Model[settings["model_part_name"].GetString()]
 
-        # Construct the component by component parameter objects
-        x_params = KratosMultiphysics.Parameters("{}")
-        y_params = KratosMultiphysics.Parameters("{}")
-        z_params = KratosMultiphysics.Parameters("{}")
+        # Get domain size
+        # Note that we check both the current model part and the root one
+        if self.model_part.ProcessInfo.Has(KratosMultiphysics.DOMAIN_SIZE):
+            domain_size = self.model_part.ProcessInfo[KratosMultiphysics.DOMAIN_SIZE]
+            if domain_size not in [2,3]:
+                root_model_part = self.model_part.GetRootModelPart()
+                if root_model_part.ProcessInfo.Has(KratosMultiphysics.DOMAIN_SIZE):
+                    domain_size = root_model_part.ProcessInfo[KratosMultiphysics.DOMAIN_SIZE]
+                else:
+                    raise Exception(f"Domain size is found neither in '{self.model_part.Name}' nor in root model part '{self.model_part.GetRootModelPart().Name}' ProcessInfo containers.")
```

Comment:
why does it have to be an exception?
Why not optional (and defaulting to 3)?

I also noticed that there are a surprisingly large number of ModelParts that don't have DOAMIN_SIZE

Reply:
basically any ModelPart that was not created directly in the solver (i.e. any aux modelpart) - and those are probably like half of them

Reply:
My thought was to somehow enforce to have this field. However, I agree that maybe it's easier, at least for the moment, to default to three to keep current behavior (see my [comment](https://github.com/KratosMultiphysics/Kratos/pull/9657#issuecomment-1039361348)).